src/LanguageServer/Protocol/Handler/Formatting/FormatDocumentOnTypeHandler.cs

```diff
@@ -72,6 +72,13 @@ public FormatDocumentOnTypeHandler(IGlobalOptionService globalOptions)
                 return [];
             }
 
+            if (SyntaxFacts.IsNewLine(request.Character[0]))
+            {
+                // When formatting after a newline is pressed, the cursor line will be blank and we do
+                // not want to remove the whitespace indentation from it. 
+                textChanges = textChanges.WhereAsArray(change => !change.Span.Contains(position));
```

Comment:
i'm not really getting the comment.  in the case of a newline, you do remove some of the changes.  could you bemore specific in the comment, perhaps with examples of the data we get back, what we *want* to filter out, and waht we *want* to keep?

Reply:
Updated!